go-controller/pkg/ovn/loadbalancer.go

```diff
@@ -1,108 +1,152 @@
 package ovn
 
 import (
-	"encoding/json"
 	"fmt"
-	"strings"
 
-	util "github.com/openvswitch/ovn-kubernetes/go-controller/pkg/util"
+	"github.com/TomCodeLV/OVSDB-golang-lib/pkg/dbtransaction"
+	"github.com/TomCodeLV/OVSDB-golang-lib/pkg/helpers"
 	"github.com/sirupsen/logrus"
 	kapi "k8s.io/api/core/v1"
 )
 
 func (ovn *Controller) getLoadBalancer(protocol kapi.Protocol) (string,
 	error) {
+	if protocol != kapi.ProtocolTCP && protocol != kapi.ProtocolUDP {
+		return "", fmt.Errorf("unknown protocol")
+	}
+
 	if outStr, ok := ovn.loadbalancerClusterCache[string(protocol)]; ok {
 		return outStr, nil
 	}
 
-	var out string
-	var err error
+	var key string
 	if protocol == kapi.ProtocolTCP {
-		out, _, err = util.RunOVNNbctlHA("--data=bare",
-			"--no-heading", "--columns=_uuid", "find", "load_balancer",
-			"external_ids:k8s-cluster-lb-tcp=yes")
-	} else if protocol == kapi.ProtocolUDP {
-		out, _, err = util.RunOVNNbctlHA("--data=bare", "--no-heading",
-			"--columns=_uuid", "find", "load_balancer",
-			"external_ids:k8s-cluster-lb-udp=yes")
-	}
-	if err != nil {
-		return "", err
+		key = "k8s-cluster-lb-tcp"
+	} else {
+		key = "k8s-cluster-lb-udp"
 	}
-	if out == "" {
-		return "", fmt.Errorf("no load-balancer found in the database")
+
+	lbs := ovn.ovnNbCache.GetMap("Load_Balancer", "uuid")
+	for _, lb := range lbs {
+		externalIds := lb.(map[string]interface{})["external_ids"]
+		if externalIds == nil {
+			continue
+		}
+
+		externalID := externalIds.(map[string]interface{})[key]
+		if externalID != nil && externalID.(string) == "yes" {
+			ovn.loadbalancerClusterCache[string(protocol)] = lb.(map[string]interface{})["uuid"].(string)
+			return lb.(map[string]interface{})["uuid"].(string), nil
+		}
 	}
-	ovn.loadbalancerClusterCache[string(protocol)] = out
-	return out, nil
+
+	ovn.loadbalancerClusterCache[string(protocol)] = ""
+	return "", nil
 }
 
 func (ovn *Controller) getDefaultGatewayLoadBalancer(protocol kapi.Protocol) string {
 	if outStr, ok := ovn.loadbalancerGWCache[string(protocol)]; ok {
 		return outStr
 	}
 
-	var gw string
-	gw, _, _ = util.RunOVNNbctlUnix("--data=bare",
-		"--no-heading", "--columns=name", "find", "logical_router",
-		"options:lb_force_snat_ip=100.64.1.2")
-	if len(gw) == 0 {
+	var defaultGateway string
+
+	gateways := ovn.ovnNbCache.GetMap("Logical_Router", "uuid")
+	for _, gateway := range gateways {
+		options := gateway.(map[string]interface{})["options"]
+		if options == nil {
+			continue
+		}
+		option := options.(map[string]interface{})["lb_force_snat_ip"]
+		if option != nil && option.(string) == "100.64.1.2" {
```

Comment:
"100.64.1.2" this looks like a magic one thing, I would suggest to store this IP in meaningfully named variable or at least write a comment what is this IP address representing